mantis/analysis/deskew.py

```diff
@@ -127,65 +128,62 @@ def deskew_data(
         - axis 0 corresponds to the scanning axis
         - axis 1 corresponds to the "tilted" axis
         - axis 2 corresponds to the axis in the plane of the coverslip
+    ls_angle_deg : float
+        angle of light sheet with respect to the optical axis in degrees
     px_to_scan_ratio : float
         (pixel spacing / scan spacing) in object space
         e.g. if camera pixels = 6.5 um and mag = 1.4*40, then the pixel spacing
         is 6.5/(1.4*40) = 0.116 um. If the scan spacing is 0.3 um, then
         px_to_scan_ratio = 0.116 / 0.3 = 0.386
-    ls_angle_deg : float
-        angle of light sheet with respect to the optical axis in degrees
     keep_overhang : bool
         If true, compute the whole volume within the tilted parallelepiped.
         If false, only compute the deskewed volume within a cuboid region.
     average_n_slices : int, optional
         after deskewing, averages every n slices (default = 1 applies no averaging)
-    order : int, optional
-        interpolation order (default 1 is linear interpolation)
-    cval : float, optional
-        fill value area outside of the measured volume (default None fills
-        with the minimum value of the input array)
+    device : str, optional
+        torch device to use for computation. Default is 'cpu'.
     Returns
     -------
     deskewed_data : NDArray with ndim == 3
         axis 0 is the Z axis, normal to the coverslip
         axis 1 is the Y axis, input axis 2 in the plane of the coverslip
         axis 2 is the X axis, the scanning axis
     """
-    if cval is None:
-        cval = np.min(np.ravel(raw_data))
-
     # Prepare transforms
-    Z, Y, X = raw_data.shape
-
     ct = np.cos(ls_angle_deg * np.pi / 180)
-    Z_shift = 0
-    if not keep_overhang:
-        Z_shift = int(np.floor(Y * ct * px_to_scan_ratio))
-
     matrix = np.array(
         [
             [
                 -px_to_scan_ratio * ct,
                 0,
                 px_to_scan_ratio,
-                Z_shift,
+                0,
             ],
-            [-1, 0, 0, Y - 1],
-            [0, -1, 0, X - 1],
+            [-1, 0, 0, 0],
+            [0, -1, 0, 0],
+            [0, 0, 0, 1],
         ]
     )
     output_shape, _ = get_deskewed_data_shape(
         raw_data.shape, ls_angle_deg, px_to_scan_ratio, keep_overhang
     )
 
-    # Apply transforms
-    deskewed_data = scipy.ndimage.affine_transform(
-        raw_data,
-        matrix,
-        output_shape=output_shape,
-        order=order,
-        cval=cval,
-    )
+    # convert to tensor on GPU
+    # convert raw_data to int32 if it is uint16
+    if raw_data.dtype == np.uint16:
+        raw_data = raw_data.astype(np.int32)
+    raw_data_tensor = torch.as_tensor(raw_data, device=device)
```

Comment:
`Affine`'s output is always `float32`, so this extra casting can be saved by always casting to float.
```suggestion
    raw_data_tensor = torch.from_numpy(raw_data.astype(np.float32)).to(device)
```

Reply:
makes sense, thanks. I double check that this doesn't require any extra GPU memory in practice